src/main/groovy/wooga/gradle/paket/base/utils/internal/DownloadAndUnpackTar.groovy

```diff
@@ -0,0 +1,56 @@
+package wooga.gradle.paket.base.utils.internal
+
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+
+class DownloadAndUnpackTar {
+    private String url;
+    private File destinationDir;
+
+    private static final String tempTarName = "temp.tgz"
+
+    DownloadAndUnpackTar(String url, File destinationDir)
+    {
+        this.url = url
+        this.destinationDir = destinationDir;
+    }
+
+    DownloadAndUnpackTar(PaketUPMWrapperReference ref, File outputDirectory)
+    {
+        this(ref.upmPackageURL, new File(outputDirectory, ref.upmPackageName))
+    }
+
+    private File getTempTarLocation()
+    {
+        return new File(this.destinationDir.getParentFile(), tempTarName);
+    }
+
+    public void exec(Project project)
+    {
+        File f = getTempTarLocation()
```

Comment:
does the temp tar file need to be downloaded to the same output directory?
You could also go with:
```
def f = File.createTempFile("prefix", "postfix")
f.deleteOnExit()
```

This would put the temp file in the system temp directory. Also `deleteOnExit` will cleanup the file when the JVM exists.

Reply:
Thanks for the tip. Happy to use this, wasn't aware of all this cool functionality :)